### PR TITLE
Add more ESLint rules and change the formatting scripts

### PR DESCRIPTION
**Pull request by lipis**
### Done

- Installed `eslint-prettier-plugin` to be able to format/fix files using ESLint and Prettier at the same time (for code files)
- All added rules are autofixable
	- [`curly`](https://eslint.org/docs/rules/curly)
	- [`no-else-return`](https://eslint.org/docs/rules/no-else-return)
	- [`no-useless-return`](https://eslint.org/docs/rules/no-useless-return)
	- [`prefer-const`](https://eslint.org/docs/rules/prefer-const)
	- [`prefer-template`](https://eslint.org/docs/rules/prefer-template)
- Updated commit hooks to fix files if autofixable before committing (it was just checking for errors before)
- Reformat all files

### TODO

- While we at it.. maybe adding more suggested rules
- What about changing the [`Bracket Spacing`](https://prettier.io/docs/en/options.html#bracket-spacing) to `false`

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -575,16 +591,17 @@ export class App extends React.Component<any, AppState> {
...
                   exportBackground: this.state.exportBackground,
                   name: this.state.name,
                   viewBackgroundColor: this.state.viewBackgroundColor,
                   scale,
                 });
+              }
             }}
             onExportToBackend={exportedElements => {
-              if (this.canvas)
+              if (this.canvas) {
```
Comment (lipis) [inline src/index.tsx]: Especially here.. we could easily miss it if making updates.

File: src/scene/data.ts [hunk 2]
```diff
@@ -190,8 +189,9 @@ export async function exportCanvas(
     scale?: number;
   },
 ) {
-  if (!elements.length)
+  if (!elements.length) {
     return window.alert(i18n.t("alerts.cannotExportEmptyCanvas"));
+  }
```
Comment (lipis) [inline src/scene/data.ts]: Confusing without the `{}`

File: src/actions/manager.tsx [hunk 3]
```diff
@@ -37,7 +37,9 @@ export class ActionManager implements ActionsManagerInterface {
         action => action.keyTest && action.keyTest(event, elements, appState),
       );
 
-    if (data.length === 0) return null;
+    if (data.length === 0) {
+      return null;
+    }
```
Comment (lipis) [inline src/actions/manager.tsx]: Easier for debugging and not polluting the diff if something will be added

File: src/components/Popover.tsx [hunk 4]
```diff
@@ -26,11 +26,11 @@ export function Popover({
 
       const viewportWidth = window.innerWidth;
       if (x + width > viewportWidth) {
-        element.style.left = viewportWidth - width + "px";
+        element.style.left = `${viewportWidth - width}px`;
       }
       const viewportHeight = window.innerHeight;
       if (y + height > viewportHeight) {
-        element.style.top = viewportHeight - height + "px";
+        element.style.top = `${viewportHeight - height}px`;
```
Comment (lipis) [inline src/components/Popover.tsx]: much better

File: src/element/collision.ts [hunk 5]
```diff
@@ -61,9 +61,8 @@ export function hitTest(
       return (
         a * tx - (px - lineThreshold) >= 0 && b * ty - (py - lineThreshold) >= 0
       );
-    } else {
-      return Math.hypot(a * tx - px, b * ty - py) < lineThreshold;
     }
+    return Math.hypot(a * tx - px, b * ty - py) < lineThreshold;
```
Comment (lipis) [inline src/element/collision.ts]: `no-else-return`

File: src/index.tsx [hunk 6]
```diff
@@ -361,10 +371,15 @@ export class App extends React.Component<any, AppState> {
...
-          else if (event.key === KEYS.ARROW_DOWN) element.y += step;
+          if (event.key === KEYS.ARROW_LEFT) {
+            element.x -= step;
+          } else if (event.key === KEYS.ARROW_RIGHT) {
+            element.x += step;
+          } else if (event.key === KEYS.ARROW_UP) {
+            element.y -= step;
+          } else if (event.key === KEYS.ARROW_DOWN) {
+            element.y += step;
+          }
```
Comment (lipis) [inline src/index.tsx]: Much easier to read.

**Comment by dwelle** [comment]
For the `prefer-const`, we can use (and should use IMO) `destructuring: "all"` to avoid the disabling:

```
"prefer-const": [
    "warn",
    {
        "destructuring": "all"
    }
],
```

**Comment by vjeux** [comment]
Ok let's merge it in. But let's not add more later. I don't think that we've had any bugs because of any of that so this is just more overhead.

**Pull request merged**